devilutionx 1.4.0 - build failure - SDL2main

As part of [ZHF for NixOS 22.11](https://github.com/NixOS/nixpkgs/issues/199919), I found that [devilutionx](https://github.com/diasurgical/devilutionX) no longer builds correctly.

### Steps To Reproduce
Steps to reproduce the behavior:
1. build *devilutionx*

### Build log
[The first time](https://hydra.nixos.org/build/194823671) it started to fail on hydra, and here's the latest [failure](https://hydra.nixos.org/build/197444853). I can reproduce it locally.

I think that this is the important part, where `SDL2::SDL2main` was not found:

```
...
-- Found Gettext: /nix/store/yzhdxmwvrcgc7i5kizfy45490169gzn5-gettext-0.21/bin/msgmerge (found version "0.21") 
-- Configuring done
CMake Error at CMakeLists.txt:243 (target_link_libraries):
  Target "devilutionx" links to:

    SDL2::SDL2main

  but the target was not found.  Possible reasons include:
    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.

CMake Error at CMake/Dependencies.cmake:73 (target_link_libraries):
  The link interface of target "DevilutionX::SDL" contains:

    SDL2::SDL2main

  but the target was not found.  Possible reasons include:
    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.

Call Stack (most recent call first):
  CMakeLists.txt:203 (include)
```

### Notify maintainers

I'm the maintainer, but I'm not sure how to fix it...

### Metadata

```console
$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.0.5, NixOS, 22.11 (Raccoon), 22.11pre422009.fdebb81f45a`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.11.0`
 - channels(karol): `""`
 - channels(root): `"home-manager, nixos"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

--- a/pkgs/development/libraries/SDL2/default.nix
+++ b/pkgs/development/libraries/SDL2/default.nix
@@ -59,11 +59,11 @@ with lib;
 
 stdenv.mkDerivation rec {
   pname = "SDL2";
-  version = "2.24.1";
+  version = "2.24.2";
 
   src = fetchurl {
     url = "https://www.libsdl.org/release/${pname}-${version}.tar.gz";
-    sha256 = "sha256-vBIViLEQUGVZjOOAeAJqQUwo6pXmbtKtq0xE2Aswnhs=";
+    sha256 = "sha256-s17wqAKwnZDtOt0NysDpWCCAQgKRT1u3sP63EPGhMp8=";
   };
   dontDisableStatic = if withStatic then 1 else 0;
   outputs = [ "out" "dev" ];

--- a/pkgs/development/libraries/SDL2/default.nix
+++ b/pkgs/development/libraries/SDL2/default.nix
@@ -59,11 +59,11 @@ with lib;
 
 stdenv.mkDerivation rec {
   pname = "SDL2";
-  version = "2.24.0";
+  version = "2.24.1";
 
   src = fetchurl {
     url = "https://www.libsdl.org/release/${pname}-${version}.tar.gz";
-    sha256 = "sha256-keTDSxdo+S05mweOFxRIxq8Yyv2nQ5h+0gZKKJVNbZc=";
+    sha256 = "sha256-vBIViLEQUGVZjOOAeAJqQUwo6pXmbtKtq0xE2Aswnhs=";
   };
   dontDisableStatic = if withStatic then 1 else 0;
   outputs = [ "out" "dev" ];

